src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -17,13 +24,38 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteSupplierCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_GOOD_NAME);
+
+        Index index;
+
         try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteSupplierCommand(index);
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteSupplierCommand.MESSAGE_USAGE), pe);
         }
+
+        DeleteSupplierCommand.DeleteSupplierGoodName deleteSupplierGoodName =
+                new DeleteSupplierCommand.DeleteSupplierGoodName();
+
+        parseGoodNamesForEdit(argMultimap.getAllValues(PREFIX_GOOD_NAME))
+                .ifPresent(deleteSupplierGoodName::setGoodNames);
+
+        return new DeleteSupplierCommand(index, deleteSupplierGoodName);
+    }
+
+    /**
+     * Parses {@code Collection<String> goodNames} into a {@code Set<GoodName>} if {@code goodNames} is non-empty.
+     */
+    private Optional<Set<GoodName>> parseGoodNamesForEdit(Collection<String> goodNames) throws ParseException {
+        assert goodNames != null;
+
+        if (goodNames.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> goodNameList = goodNames.size() == 1 && goodNames.contains("") ? Collections.emptyList()
+                : goodNames;
```

Comment:
Can you add comments for this code, or refactor it into a method with an explanatory name?